crates/primitives/chain_config/src/rpc_version.rs

```diff
@@ -0,0 +1,153 @@
+use std::hash::Hash;
+use std::str::FromStr;
+
+lazy_static::lazy_static! {
+    pub static ref SUPPORTED_RPC_VERSIONS: Vec<RpcVersion> = vec![
+        RpcVersion::RPC_VERSION_0_7_1,
+    ];
+}
+
+#[derive(Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, serde::Serialize, serde::Deserialize, Hash)]
+pub struct RpcVersion([u8; 3]);
+
+#[derive(thiserror::Error, Debug, PartialEq)]
+pub enum RpcVersionError {
+    #[error("Invalid number in version")]
+    InvalidNumber(#[from] std::num::ParseIntError),
+    #[error("Too many components in version: {0}")]
+    TooManyComponents(usize),
+    #[error("Invalid request path specified, could not extract a version")]
+    InvalidPathSupplied,
+    #[error("Invalid version specified")]
+    InvalidVersion,
+    #[error("Unsupported version specified")]
+    UnsupportedVersion,
+}
+
+impl RpcVersion {
+    pub const fn new(major: u8, minor: u8, patch: u8) -> Self {
+        RpcVersion([major, minor, patch])
+    }
+
+    pub fn from_request_path(path: &str) -> Result<Self, RpcVersionError> {
+        let parts: Vec<&str> = path.split('/').collect();
```

Comment:
can't we use a regex to simplify this function?

Reply:
I try to avoid regexes for performance & safety (like overly permissive regex) - @cchudant wdyt?

Imo it's fine like this but can do it